flink-runtime/src/main/java/org/apache/flink/runtime/operators/AbstractOuterJoinDriver.java

```diff
@@ -0,0 +1,166 @@
+		
+		final MutableObjectIterator<IT1> in1 = this.taskContext.getInput(0);
+		final MutableObjectIterator<IT2> in2 = this.taskContext.getInput(1);
+
+		// get the key positions and types
+		final TypeSerializer<IT1> serializer1 = this.taskContext.<IT1>getInputSerializer(0).getSerializer();
+		final TypeSerializer<IT2> serializer2 = this.taskContext.<IT2>getInputSerializer(1).getSerializer();
+		final TypeComparator<IT1> comparator1 = this.taskContext.getDriverComparator(0);
+		final TypeComparator<IT2> comparator2 = this.taskContext.getDriverComparator(1);
+		
+		final TypePairComparatorFactory<IT1, IT2> pairComparatorFactory = config.getPairComparatorFactory(
+				this.taskContext.getUserCodeClassLoader());
+		if (pairComparatorFactory == null) {
+			throw new Exception("Missing pair comparator factory for outer join driver");
+		}
+
+		ExecutionConfig executionConfig = taskContext.getExecutionConfig();
+		boolean objectReuseEnabled = executionConfig.isObjectReuseEnabled();
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Outer Join Driver object reuse: " + (objectReuseEnabled ? "ENABLED" : "DISABLED") + ".");
+		}
+
+		// create and return outer join iterator according to provided local strategy.
+		if (objectReuseEnabled) {
+			setReusingOuterJoinIterator(ls, in1, in2, serializer1, comparator1, serializer2, comparator2, pairComparatorFactory,
+					memoryManager, ioManager, numPages);
+		} else {
+			setNonReusingOuterJoinIterator(ls, in1, in2, serializer1, comparator1, serializer2, comparator2, pairComparatorFactory,
+					memoryManager, ioManager, numPages);
+		}
+		
+		this.outerJoinIterator.open();
+		
+		if (LOG.isDebugEnabled()) {
+			LOG.debug(this.taskContext.formatLogString("outer join task iterator ready."));
+		}
+	}
+
+	@Override
+	public void run() throws Exception {
+		final FlatJoinFunction<IT1, IT2, OT> joinStub = this.taskContext.getStub();
+		final Collector<OT> collector = this.taskContext.getOutputCollector();
+		final JoinTaskIterator<IT1, IT2, OT> outerJoinIterator = this.outerJoinIterator;
+
+		while (this.running && outerJoinIterator.callWithNextKey(joinStub, collector));
+	}
+
+
+	@Override
+	public void cleanup() throws Exception {
+		if (this.outerJoinIterator != null) {
+			this.outerJoinIterator.close();
+			this.outerJoinIterator = null;
+		}
+	}
+
+	@Override
+	public void cancel() {
+		this.running = false;
+		if (this.outerJoinIterator != null) {
+			this.outerJoinIterator.abort();
+		}
+	}
+
+	protected abstract void setReusingOuterJoinIterator(DriverStrategy driverStrategy, MutableObjectIterator<IT1> in1,
+														MutableObjectIterator<IT2> in2, TypeSerializer<IT1> serializer1,
+														TypeComparator<IT1> comparator1, TypeSerializer<IT2> serializer2,
+														TypeComparator<IT2> comparator2, TypePairComparatorFactory<IT1, IT2> pairComparatorFactory,
+														MemoryManager memoryManager, IOManager ioManager, int numPages) throws Exception;
```

Comment:
I'm not a big fan of different indentation schemes as it is here the case for the first parameter line and the rest, especially if you use tabs. IMO it would be better to also indent the first parameters in a new line with the same indentation as the other parameters. Moreover, putting each parameter in a new line allows to easily add and remove parameters.